### ISSUE TITLE
support DNS-over-TLS

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#1471]
we (well, twisted?) should totally support DNS-over-TLS:

https://security.googleblog.com/2019/01/google-public-dns-now-supports-dns-over.html

**Comment by oberstet** [comment]
both google (quad 8) and cloudflare (quad 1) support it already. neat

https://developers.cloudflare.com/1.1.1.1/dns-over-tls/

**Comment by meejah** [comment]
Is this different from DOH (DNS Over HTTPS)? https://en.wikipedia.org/wiki/DNS_over_HTTPS

**Comment by oberstet** [comment]
yes, DNS-over-TLS and DNS-over-HTTPS are 2 different approaches aiming at the same issue (privacy) as far as I can see:

* https://tools.ietf.org/html/rfc8484
* https://tools.ietf.org/html/rfc7858
